### ISSUE TITLE
Invoke local and invoke hang when connected to a debugger in vs code

### ISSUE THREAD

**Issue by ricardolpd** [serverless/serverless#4467]
# Bug Report

## Description

For bug reports:
* What went wrong?
I usually install serverless in relation to my project and then i set vs code to start the serverless file in the node_modules, which seems to work up to now with any command with exception of invoke and invoke local, in which sls just hangs in the plugin without executing any function.

* What did you expect should have happened?
With any other command i can follow the execution.
* What was the config you used?
In visual studio code i usually set my launch.json to something like this:
`{
        "type": "node",
        "request": "launch",
        "name": "launch cryptonator aggregator",
        "program": "${workspaceRoot}/node_modules/serverless/bin/serverless",
        "cwd": "${workspaceRoot}",
        "protocol": "inspector",
        "args": [
            "invoke",
            "local",
            "-f",
            "foo-function"
    ],
}`
which results in something like the following in the debugger:

> node --inspect-brk=49604 node_modules/serverless/bin/serverless deploy -f foo-function

* What stacktrace or error message from your provider did you see?
No errors it just hangs in the following like in the serverless/lib/aws/invokeLocal/index.js:
```
stdin().then(input => {
   this.options.data = input;
   resolve();
});

```
At the moment i done the following:
```
const debug = typeof v8debug === ‘object’ || /–debug|–inspect/.test(process.execArgv.join(’ '));
if (debug) {
   this.serverless.cli.log(‘skipping stdin pipping’);
   resolve();
} else {
   stdin().then(input => {
      this.options.data = input;
      resolve();
   });
}
```
- Has anyone else found this behaviour? 
- Would it be possible to fix?
Is the solution above too hacky (i personally think so) does anyone have a possible better solution? Happy to put something together.
## Additional Data
Serverless Framework Version you're using: 1.23.0 
It happens with both node v8.7.0 and v6.1.0.
Operating System: MAC OS 10.12.6

**Comment by ricardolpd** [comment]
@RafalWilinski sorry to but in, why is this marked as an enchancement, from where i stand it looks like a defect. Could you please clarify why is it not marked as defect.

**Comment by RafalWilinski** [comment]
My bad, sorry! Fixed now

**Comment by tom-thorne** [comment]
I fixed the same issue by setting console to IntegratedTerminal...
```
{
    "version": "0.2.0",
    "configurations": [
        {
            "type": "node",
            "request": "launch",
            "name": "Launch Program",
            "program": "/usr/local/bin/sls",
            "args": [
                "invoke", "local", "-f", "hello"
            ],
            // required otherwise serverless thinks stdin is a stream and tries to read options from it
            "console": "integratedTerminal",
        }
    ]
}
```

**Comment by jaydp17** [comment]
@tom-thorne thanks for the workaround, but by doing this, we can't use the Debug Console to evaluate expressions 😕

**Comment by jaydp17** [comment]
@ricardolpd as shown in https://hackernoon.com/running-and-debugging-aws-lambda-functions-locally-with-the-serverless-framework-and-vs-code-a254e2011010, if we pass `--data {}` or the data you wanna provide, it works as expected without `"console": "integratedTerminal",`.

So for @ricardolpd it'd look something like
```json
{
        "type": "node",
        "request": "launch",
        "name": "launch cryptonator aggregator",
        "program": "${workspaceRoot}/node_modules/serverless/bin/serverless",
        "cwd": "${workspaceRoot}",
        "protocol": "inspector",
        "args": [
            "invoke",
            "local",
            "-f",
            "foo-function",
            "--data",
            "{}",
    ],
}
```

**Comment by ricardolpd** [comment]
@jaydp17 yeah i know that but i need to pass data to the function to evaluate it unfortunately, i am using at the moment @tom-thorne workaround, but it is not great for the reasons you mention, maybe serverless could add a flag to disable that would disable the use of stdin... not sure

**Comment by kamiljano** [comment]
any news on this issue?
I'm experiencing the same problem with serverless 1.38.0 and Node 10.15.2 (as of now the latest versions of both)
